### PR TITLE
switch to axios and bringing lots of things up to modern times

### PR DESCRIPTION
**Pull request by zenion**

### CODE REVIEW

**Review by noticeeverything — APPROVED** [review]
Looks like a big improvement. 

I would think about separating the types/interfaces from the main file. 

An ongoing JavaScript/TypeScript conundrum is whether to prefix interface names with `I`. I  tend to lean more towards the avoid-it camp, but don't have super strong feelings on it. It is a best practice to name interfaces like that in C#, for whatever that's worth.

Yet another ongoing debate is whether to prefix private methods with `_`. I actually prefer that because it's easy to see that a member is private based on the name. Based on what I've read on the interwebs, I'm in the minority on that one.

In my own projects I try to avoid implicit or explicit `any` typings. I usually set my eslint to error for that. The types I'm writing for the nest api may be appropriate to extract into a published lib so projects like this can access them.

**Comment by zenion** [comment]
Library is single file so seperating types does nothing except makes me
have to click around in IDE at this point. Also not enough time to dedicate
to typing all the shit coming back from egnytes shitty service.. maybe one
day lol.

We actually could move to ES2020 private variables but in practice they
don't quite have full feature parity with typescript so I just avoided it
and did what I normally do which is nonruntime.

**Comment by CalebAlbers** [comment]
I haven't reviewed the code yet, but I agree - prefixing I on interfaces is against TypeScript's best practice manual. I generally have a distaste for any form of Hungarian notation though.

**Comment by zenion** [comment]
I agree with the interface names thing btw... You know me, whatever the community is doing is what i'd prefer as well. In this case it's just the way i did it when i made these 3 years ago and knew nothing about what i was doing really.

File: package.json [hunk 1]
```diff
@@ -25,29 +26,42 @@
     "url": "https://github.com/zenion/egnyte-resellers-node"
   },
   "dependencies": {
-    "got": "^11.3.0"
+    "axios": "^0.21.1",
+    "got": "^11.8.1"
```
Comment (CalebAlbers) [inline package.json]: I think `got` was left in by accident.

File: src/index.ts [hunk 2]
```diff
@@ -225,56 +225,64 @@ class Egnyte {
...
-      headers: {
-        Cookie: authCookie,
-        'Content-Type': 'application/json',
-        'X-Requested-With': 'XMLHttpRequest'
+  /**
+   * Updates a plan's power user licensing count. This will result in a billing change from egnyte. Must be increased in increments of 5
+   * @param planId the planId to update licensing for
+   * @param newTotalLicenses the new total in increments of 5. Use getPlans() to find current total for given plan
+   */
+  async UpdatePowerUserLicensing(planId: string, newTotalLicenses: number) {
```
Comment (CalebAlbers) [inline src/index.ts:267]: Does Egnyte toss a friendly error upstream if the `newTotalLicenses` number isn't a multiple of 5?

If so, probably easiest to leave as-is. If not, we could potentially add a type guard to check `newTotalLicenses % 5`

Comment (zenion) [reply src/index.ts:267]: yeah it errors from service. I felt like the description in IDE would be useful enough but feel free to improve things guys.

Comment (CalebAlbers) [reply src/index.ts:267]: As long as it errors from the service intelligently, I don't think it's worth the effort. Was only thinking in the event it was some really obscure error that required reverse engineering to figure out.

**Pull request merged**